src/components/ScholarshipDetailCard.jsx

```diff
@@ -63,6 +67,36 @@ export default function ScholarshipDetailCard({ scholarship }) {
     eligibility,
   } = scholarship.data;
 
+  const shareDialog = {
+    showShareDialog: true,
+    shareTitle: `${amount?.toString()} - ${name} | ${BRAND_NAME}`,
+    shareURL: `https://${window.location.hostname}/scholarships/${scholarship.id}`,
+  };
+
+  const shareText = `${
+    shareDialog.shareTitle
+  } \n ${deadline?.toLocaleDateString()}\n`;
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars, no-unused-vars
+  const shareFn = () => {
+    if (navigator.share) {
+      navigator
+        .share({
+          title: shareDialog.shareTitle,
+          url: shareDialog.shareURL,
+          text: shareText,
+        })
+        // eslint-disable-next-line no-console
+        .then(() => console.log('Thanks for sharing!'))
+        // eslint-disable-next-line no-console
+        .catch(console.error);
+    } else {
+      history.replace({
+        state: { shareDialog },
+      });
+    }
```

Comment:
I think to keep things simple we should use the [share `data` object parameter](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/share#parameters) `navigator.share` expects and set `showShareDialog` as a separate state variable. E.g.:

```js
  const shareFn = () => {
    const data = { url: ..., text: ..., title: ...};
    if (navigator.share) {
      navigator.share(data)
        // eslint-disable-next-line no-console
        .then(() => console.log('Thanks for sharing!'))
        // eslint-disable-next-line no-console
        .catch(console.error);
    } else {
      history.replace({
        state: { showShareDialog: true, shareData: data },
      });
    }
  };
```